x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -564,14 +683,16 @@ public DatafeedConfig build() {
                 throw invalidOptionValue(TYPES.getPreferredName(), types);
             }
 
-            validateAggregations();
-            setDefaultChunkingConfig();
+            AggregatorFactories.Builder parsedAggs = lazyAggParser.apply(aggregations, id);
```

Comment:
This is sort of tricky, when a user calls `GET` for a DatafeedConfig, `DatafeedConfig.Builder#build()` is called, I think...

If it is (still reading through to verify), I may have to figure out how to do these validations WITHOUT parsing the aggregations.

Reply:
ugh, yeah, verified by hijacking a yml test and forcing to always use lenient parsing:

```
"Test put datafeed with invalid query":
  - do:
      xpack.ml.put_datafeed:
        datafeed_id: test-datafeed-1
        body:  >
          {
            "job_id":"datafeeds-crud-1",
            "indexes":["index-foo"],
            "types":["type-bar"],
            "query":{"match_all_misspelled":"thing"}
            "aggregations": {"max":"invalid"}
          }
  - do:
      xpack.ml.get_datafeeds:
        datafeed_id: test-datafeed-1

  - match: { datafeeds.0.datafeed_id: "test-datafeed-1" }
  - match: { datafeeds.0.query.match_all_misspelled: "thing" }
  - match: { datafeeds.0.aggregations.max: "invalid" }
```

This fails :(. Gotta try and do the agg validations without parsing them

Reply:
> If it is (still reading through to verify), I may have to figure out how to do these validations WITHOUT parsing the aggregations.

I would like the builders not to validate when `build()` is called instead they only need to validated when created or the datafeed is started. With the config in the index the current implementation validates every time the document is read.  #34899 is on my wishlist

Reply:
Thanks for the feedback y'all, I will get these concerns addressed soon.